### PR TITLE
[flang][openacc] Support multiple device_type when lowering

### PR DESCRIPTION
**Pull request by clementval**
routine, data, parallel, serial, kernels and loop construct all support the device_type clause. This clause takes a list of device_type. Previously the lowering code was assuming that the list s a single item. This PR updates the lowering to handle any number of device_types.

### CODE REVIEW

File: flang/test/Lower/OpenACC/acc-device-type.f90 [hunk 1]
```diff
@@ -40,5 +40,9 @@ subroutine sub1()
 
 ! CHECK: acc.parallel num_gangs({%c2{{.*}} : i32}, {%c1{{.*}} : i32, %c1{{.*}} : i32, %c1{{.*}} : i32} [#acc.device_type<nvidia>])
 
+  !$acc parallel device_type(nvidia, default) num_gangs(1, 1, 1)
+  !$acc end parallel
+
+! CHECK: acc.parallel num_gangs({%c1_i32_9 : i32, %c1_i32_10 : i32, %c1_i32_11 : i32} [#acc.device_type<nvidia>], {%c1_i32_9 : i32, %c1_i32_10 : i32, %c1_i32_11 : i32} [#acc.device_type<default>])
```
Comment (vzakhari) [inline flang/test/Lower/OpenACC/acc-device-type.f90:46]: Please do not hardcode the value names in the check pattern.

**Review by vzakhari — APPROVED** [review]
LGTM with minor remark for the LIT test.

File: flang/test/Lower/OpenACC/acc-device-type.f90 (hunk 1 above)
Comment (clementval) [reply flang/test/Lower/OpenACC/acc-device-type.f90:46]: My bad I just copy paste it and forgot to update the ssa values. I'll update that.

Comment (clementval) [inline flang/test/Lower/OpenACC/acc-device-type.f90:46]:
```suggestion
! CHECK: acc.parallel num_gangs({%c1{{.*}} : i32, %c1{{.*}} : i32, %c1{{.*}} : i32} [#acc.device_type<nvidia>], {%c1{{.*}} : i32, %c1{{.*}} : i32, %c1{{.*}} : i32} [#acc.device_type<default>])
```

**Pull request merged**